RadixWallet/Core/DesignSystem/Components/Hint.swift

```diff
@@ -1,9 +1,16 @@
 
 // MARK: - Hint
 public struct Hint: View, Equatable {
-	public struct ViewState: Equatable {
+	public struct ViewState: Equatable, Identifiable {
 		public let kind: Kind
 		public let text: Text?
+		public let id: UUID
```

Comment:
In general this is pretty risky, you need to be very careful so that you only create these once when using UUID as ID

Reply:
Forgot adding a comment on this class, but it was related to this in particular so will answer you here.

We are using `Hint` all over the app as a `View` that gets sent by parameter. This isn't very optimal as it is hard to compare, iterate, etc. Similar thing happens with the `let text: Text?` that the `Hint.ViewState` has.

I would recommend moving into a `ViewState` that instead has a `let text: String?`. Then, it is the Hint's `var body: some View` responsibility to create the `Text` from the `String`.

Same approach for the `Hint`: instead of sending a `Hint`, send a `Hint.ViewState` (example on `AppTextField`).

I didn't want to get into this change because it involves quite a lot of classes unrelated to this PR (which is big enough already). But if you like the approach I am happy to work on it.

Reply:
I am sure your changes to Hint are thought through, I was only really talking about the practice of using UUID as ID for Identifiable. When I say risky I meant that SwiftUI (and also TCA) may sometimes create new copies of things, and if we happen to then also give them a new UUID, the ID system will break down. So it's important to make sure that we are really only creating the UUID once, for example when we fetch something from an API or something. Definitely not in say a computed property.

Anyway, that was just a general comment, I don't know if you are doing that here, and in any case even if things did go wrong here, it wouldn't be a big issue, at most a visual glitch. But for example in TransactionHistory, I use UUID in one place, and there it would indeed be pretty bad if we accidentally created a new ID for some entry.

Reply:
Yes, I total see where you come from and why you highlighted the risks of using an UUID for something that isn't actually a model. Let me explain why I proposed this Hint refactor solution.

With the possibility of having multiple hints chained (as we discussed on that thread with Aftab and Matt), I needed to add the `UUID` in order to allow looping with a `ForEach` an array of `[Hint.ViewState]`. Given that its state is defined by a `View` (a `Text`), it isn't possible to identify them. The cleanest solution would be changing the entire View and uses of it, but as the current solution isn't critical (as you say, worst case it has a UI glitch), I wanted to ask for opinions before diving into it.

Reply:
The downside with using index is that the view will not behave correctly if the array ever changes. To be precise, if the change is "add new rows at the end" it will work, but if you insert some new element at say row 3, SwiftUI will not know that this is different from the old index 3. Sometimes that will give incorrect results.